registry/retriever/extrinsic_retriever_live_test.go

```diff
@@ -0,0 +1,198 @@
+//go:build live
+
+package retriever
+
+import (
+	"errors"
+	"log"
+	"sync"
+	"testing"
+
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client/v4"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/rpc/chain/generic"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/scale"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+var (
+	extTestChains = []testFnProvider{
+		&testChain[
+			types.MultiAddress,
+			types.MultiSignature,
+			generic.DefaultPaymentFields,
+		]{
+			url: "wss://fullnode.parachain.centrifuge.io",
+		},
+		&testChain[
+			types.MultiAddress,
+			types.MultiSignature,
+			generic.DefaultPaymentFields,
+		]{
+			url: "wss://rpc.polkadot.io",
+		},
+		&testChain[
+			types.MultiAddress,
+			types.MultiSignature,
+			generic.PaymentFieldsWithAssetID,
+		]{
+			url: "wss://statemint-rpc.polkadot.io",
+		},
+		&testChain[
+			types.MultiAddress,
+			AcalaMultiSignature,
+			generic.DefaultPaymentFields,
+		]{
+			url: "wss://acala-rpc-0.aca-api.network",
+		},
+		&testChain[
+			[20]byte,
+			[65]byte,
+			generic.DefaultPaymentFields,
+		]{
+			url: "wss://wss.api.moonbeam.network",
+		},
+	}
+)
+
+func TestLive_ExtrinsicRetriever_GetExtrinsics(t *testing.T) {
+	t.Parallel()
+
+	var wg sync.WaitGroup
+
+	for _, c := range extTestChains {
+		c := c
+
+		wg.Add(1)
+
+		go c.GetTestFn()(&wg)
+	}
+
+	wg.Wait()
+}
+
+type testFn func(wg *sync.WaitGroup)
+
+type testFnProvider interface {
+	GetTestFn() testFn
+}
+
+type testChain[
+	A any,
+	S any,
+	P any,
+] struct {
+	url string
+}
+
+func (t *testChain[A, S, P]) GetTestFn() testFn {
+	return func(wg *sync.WaitGroup) {
+		defer wg.Done()
+
+		testURL := t.url
+
+		api, err := gsrpc.NewSubstrateAPI(testURL)
+
+		if err != nil {
+			log.Printf("Couldn't connect to '%s': %s\n", testURL, err)
+			return
+		}
+
+		chain := generic.NewChain[A, S, P, *generic.SignedBlock[A, S, P]](api.Client)
```

Comment:
Yes, there's an example of it in the `extrinsic_retriever_live_test.go` file:

```
		chain := generic.NewChain[A, S, P, *generic.SignedBlock[A, S, P]](api.Client)
```

It would've looked better if we were able to do something like:

```
		chain := generic.NewChain[*generic.SignedBlock[A, S, P]](api.Client)
```

but since `A,S,P` are generics themselves, they have to be repeated at the beginning, making the whole thing look extra complicated, when in reality `A,S,P` are only used for the 4th param which is `B`, the `*generic.SignedBlock`.

This might be an overkill though, initially, I built this `generic.Chain` with only one generic parameter, either the full block using for retrieval OR the signature type that a `generic.Block` would have. The current option provides the most amount of flexibility despite its looks...

Lemme know what's your opinion on this, we can look at the initial versions that I made for this together and make a decision, unfortunately I did not think about saving those somewhere so we'll have to "build" them again from scratch.

Reply:
Added a default constructor for the generic chain client.

Reply:
Done.